lang/extract_json_strings.py

```diff
@@ -1008,6 +1008,12 @@ def extract(item, infilename):
     if "use_action" in item:
         extract_use_action_msgs(outfile, item["use_action"], item.get("name"), kwargs)
         wrote = True
+    if "conditional_names" in item:
+        kwargs["format_strings"] = True
+        for cname in item["conditional_names"]:
+            c = "Conditional name for {} when {} matches {}".format(name, cname["type"], cname["condition"])
+            writestr(outfile, cname["name"], comment=c, **kwargs)
```

Comment:
You should add `new_pl_fmt=True` to the parameter list to make sure a plural name is always generated.